### PR TITLE
New HDAT PNOR parition addition

### PR DESCRIPTION
**Pull request by nagendraIBM**
1> Defined macros in config file for the binary files that would be used for making HDAT partition.
2> Added changes to pick up the hdat binaries for  buillding HDAT pnor partition while making pnor using create_pnor_image.pl.

### CODE REVIEW

**Comment by stewart-ibm** [comment]
Can one of the admins verify this patch for testing?

**Comment by stewart-ibm** [comment]
nagendraIBM <notifications@github.com> writes:
> 1> Defined macros in config file for the binary files that would be used for making HDAT partition.
> 2> Added changes to pick up the hdat binaries for  buillding HDAT pnor
> partition while making pnor using create_pnor_image.pl.

What is the purpose of this partition?

What goes in it? What consumes it? Does it need to be signed?

-- 
Stewart Smith
OPAL Architect, IBM.

File: openpower/package/openpower-pnor/Config.in [hunk 1]
```diff
@@ -149,3 +149,15 @@ config BR2_OPENPOWER_PNOR_XZ_ENABLED
 config BR2_BUILD_PNOR_SQUASHFS
         bool "Create a virtual PNOR image along with the standard image"
         default y
+
+config BR2_OPENPOWER_HDAT_BIN_FILENAME
+       string " Name of openpower HDAT binary file"
```
Comment (ajdlinux) [inline openpower/package/openpower-pnor/Config.in]: Drop the space at the start of string here

File: openpower/package/openpower-pnor/Config.in [hunk 2]
```diff
@@ -149,3 +149,15 @@ config BR2_OPENPOWER_PNOR_XZ_ENABLED
...
+       default "hdat.bin"
+       help
+            String used to define name of the HDAT binary file
+config BR2_OPENPOWER_HDAT_ECC_FILENAME
+       string " Name of openpower HDAT binary file with ecc"
+       default "hdat.bin.ecc"
+       help
+            String used to define name of ECC protected HDAT binary file
+
```
Comment (ajdlinux) [inline openpower/package/openpower-pnor/Config.in]: Make indentation consistent with rest of file

File: openpower/package/openpower-pnor/openpower-pnor.mk [hunk 3]
```diff
@@ -92,6 +92,7 @@ define OPENPOWER_PNOR_INSTALL_IMAGES_CMDS
             -openpower_version_filename $(OPENPOWER_PNOR_VERSION_FILE) \
             -wof_binary_filename $(OPENPOWER_MRW_SCRATCH_DIR)/$(BR2_WOFDATA_FILENAME) \
             -memd_binary_filename $(OPENPOWER_MRW_SCRATCH_DIR)/$(BR2_MEMDDATA_FILENAME) \
+			-hdat_binary_filename $(BR2_OPENPOWER_HDAT_BIN_FILENAME) \
```
Comment (ajdlinux) [inline openpower/package/openpower-pnor/openpower-pnor.mk]: Keep indentation consistent with rest of file

**Comment by dcrowell77** [comment]
The purpose of this partition is to contain some HDAT-specific data mined from the MRW.  Specifically it will include GPU and slotmap info.  That info should probably be part of a commit message though... ;-)

Yes it will need to be signed just like the other MRW-derived data.

**Comment by ajdlinux** [comment]
This is dependent on https://github.com/open-power/pnor/pull/82. There's also some corresponding hostboot changes in the works.

@dcrowell77 I assume that we'll see the code to generate the hdat binary at some point relatively soon? At some point I need to try using HDAT to get my NPU/OpenCAPI device info. :)

**Comment by dcrowell77** [comment]
Yep, getting the partition in place is a pre-req to adding the binary in.  @nagendraIBM should be covering all of it.

**Comment by ajdlinux** [comment]
What's the status of this?

**Pull request closed without merging**